File: primeFactors.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
  int input;
  cin >> input;
  vector<int> factors;
  int y = 2;
  
  while(y*y <= input){
    if(input % y == 0){
      factors.push_back(y);
      input /= y;
    }else{
      y++;
    }
  }
  if(input > 1){
    factors.push_back(input);
  }
  
  //print the factors
  for (int i = 0; i < factors.size(); ++i){
    cout << factors[i] << " ";
  }
  cout << endl;
  return 0;
}

File: primeFactors.rb
# -*- encoding : utf-8 -*-
input = gets.chomp.to_i
factors = Array.new
y = 2
while y*y <= input do
  if input % y == 0
    factors.push(y)
    input /= y
  else
    y = y+1
  end
end
if input > 1
  factors.push(input)
end
factors.each{ |x| print "%i " % [x] }
puts
